### ISSUE TITLE
Violations go undetected

### ISSUE THREAD

**Issue by RieksJ** [AmpersandTarski/Ampersand#711]
Compile the following script into a prototype:
~~~
CONTEXT Bug

geitCounter   :: Geit   * GeitCounter   [UNI]   REPRESENT GeitCounter   TYPE INTEGER
schaapCounter :: Schaap * SchaapCounter [UNI]   REPRESENT SchaapCounter TYPE INTEGER

ROLE ExecEngine MAINTAINS "Initialize GeitCounters"
RULE "Initialize GeitCounters": I[Geit] |- geitCounter;V
VIOLATION (TXT "{EX} InsPair;geitCounter;Geit;", SRC I, TXT ";GeitCounter;3")

ROLE ExecEngine MAINTAINS "Initialize SchaapCounters"
RULE "Initialize SchaapCounters": I[Schaap] |- schaapCounter;schaapCounter~
VIOLATION (TXT "{EX} InsPair;schaapCounter;Schaap;", SRC I, TXT ";SchaapCounter;5")

INTERFACE Geiten:  V[SESSION*Geit]   CRuD BOX [ Geit:   I cRud, Number: geitCounter   cRud]
INTERFACE Schapen: V[SESSION*Schaap] CRuD BOX [ Schaap: I cRud, Number: schaapCounter cRud]

POPULATION Geit   CONTAINS [ "Geit1" ]
POPULATION Schaap CONTAINS [ "Schaap1" ]

ENDCONTEXT
~~~

Then, run the installer for the prototype. This produces the following error message:
![image](https://user-images.githubusercontent.com/8522589/31673755-54b4379a-b360-11e7-9c32-435540e2b669.png)

This is a bug, since the 'geit'-errors should have been handled in the same way as the 'schaap' errors.

**Comment by stefjoosten** [comment]
Beautiful bug report, Rieks! It tells the whole story. 

Apparently something is wrong with the `V` in relation to its technical type.

I tried to reproduce this bug using Ampersand-v3.8.6 [RAP3production:73d05d5]. All I got was a perfectly working application:
![afbeelding](https://user-images.githubusercontent.com/275768/32263974-9b7f5f24-bedc-11e7-9d99-136122282a6b.png)
I even checked the database. It contains one `Geit` and one `Schaap`. So no success in reproducing the bug here. Let me try it on the most recent version on the development branch ...

**Comment by RieksJ** [comment]
This is weird. Bummer that I did not specify the &-version...

When reviewing the error, I noticed that there are two violations, one for Geit1 and another for Geit2.
That means that the error message does not apply to the source code I provided, at least not exclusively.

So I redid the bugtest with Ampersand-v3.8.8 [development:4654c6f].
This worked like one would expect. Therefore I'll close the issue.
